modules/nf-core/bbmap/bbnorm/main.nf

```diff
@@ -0,0 +1,45 @@
+process BBMAP_BBNORM {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::bbmap=39.01 pigz=2.6"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-008daec56b7aaf3f162d7866758142b9f889d690:e8a286b2e789c091bac0a57302cdc78aa0112353-0':
+        'quay.io/biocontainers/mulled-v2-008daec56b7aaf3f162d7866758142b9f889d690:e8a286b2e789c091bac0a57302cdc78aa0112353-0' }"
+
+    input:
+    tuple val(meta), path(fastq)
+
+    output:
+    tuple val(meta), path("*.fastq.gz"), emit: bam
+    tuple val(meta), path("*.log")     , emit: log
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    input  = meta.interleaved ? "in=${fastq}" : "in=${fastq[0]} in2=${fastq[1]}"
+    output = meta.interleaved ? "out=${prefix}.fastq" : "out1=${prefix[0]}.fastq in2=${prefix[1]}.fastq"
+
+    """
+    bbnorm.sh \\
+        $input \\
+        out=${prefix}.fastq \\
```

Comment:
Can't the tool write gzipped files directly? Is this how it's done in the other bbmap modules?

Reply:
I actually don't know. in bbduk the output is a *.gz so I guess it can take it. I can try.